### PR TITLE
Refactoring of get etcd tests.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Refactoring of get etcd tests: introduced new generic resttest, TestGet; Converted all etcd tests to use it.

### CODE REVIEW

File: pkg/tools/fake_etcd_client.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func (f *FakeEtcdClient) Get(key string, sort, recursive bool) (*etcd.Response,
 	result := f.Data[key]
 	if result.R == nil {
 		if _, ok := f.expectNotFoundGetSet[key]; !ok {
-			f.t.Fatalf("data for %s was not defined prior to invoking Get", key)
+			f.t.Logf("data for %s was not defined prior to invoking Get", key)
```
Comment (wojtek-t) [inline pkg/tools/fake_etcd_client.go]: Why did you change this one? I think it makes sense to have it, because otherwise it may return random garbage.

Comment (jszczepkowski) [inline pkg/tools/fake_etcd_client.go]: We are not returning random garbage, but f.NewError(EtcdErrorCodeNotFound) error. I want to handle the error in my test (I query for non-existing object), but due to f.t.Fatalf the test was dying with assertion.

Comment (wojtek-t) [inline pkg/tools/fake_etcd_client.go]: I see - thanks for explanation.

**Comment by wojtek-t** [comment]
LGTM
